homeassistant/components/synology_dsm/__init__.py

```diff
@@ -63,15 +62,7 @@ async def async_setup(hass, config):
 
 async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
     """Set up Synology DSM sensors."""
-    host = entry.data[CONF_HOST]
-    port = entry.data[CONF_PORT]
-    username = entry.data[CONF_USERNAME]
-    password = entry.data[CONF_PASSWORD]
-    unit = hass.config.units.temperature_unit
-    use_ssl = entry.data[CONF_SSL]
-    device_token = entry.data.get("device_token")
-
-    api = SynoApi(hass, host, port, username, password, unit, use_ssl, device_token)
+    api = SynoApi(hass, entry)
```

Comment:
Consider adding an update listening and reloading the config entry when it changes.  This would make this change quite a bit smaller.

```
undo_listener = entry.add_update_listener(_async_update_listener)
```
...
```
async def _async_update_listener(hass: HomeAssistant, entry: ConfigEntry):
    """Handle options update."""
    await hass.config_entries.async_reload(entry.entry_id)
```

Example in `nut` as well.

Reply:
I actually wants to add 3 other options after that, to take care of `CONF_DISKS`, `CONF_VOLUMES`, and the new security advisor API added in 0.8.0.

So I think I should keep it here

Reply:
Ideally the `hass.config_entries.async_forward_entry_unload` and `hass.config_entries.async_forward_entry_setup`
stay inside `async_unload_entry` and `async_setup_entry` as it make it harder to review because the reviewer has to go find where they are being called.